plugins/modules/lastlogins.py

```diff
@@ -166,7 +166,13 @@ def run_module():
     # last -i shows 0.0.0.0 for local logins
     allowed_ips = ['0.0.0.0']
     if module.params['allowed_ips'] is not None:
-        allowed_ips.extend(module.params['allowed_ips'])
+        for allowed_ip in module.params['allowed_ips']:
+            allowed_ips.extend([str(ip) for ip in ipaddress.IPv4Network(allowed_ip)])
```

Comment:
I have two issues with this line:
- It does not work with IPv6 at all; you're explicitly creating an `IPv4Network` object here.
- This populates a list of all individual IP addresses in the provided CIDR range. This smells like a resource hog, both in generating this list, and in iterating it for matches against the login IPs. I'd already call that a stretch for smaller IPv4 ranges such as `/24`, but it's pretty much prohibitive for larger v4 ranges (e.g. `10.0.0.0/8`), and even more so in IPv6, where you'd have to look hard to find subnets smaller than `/64`.